app/models/sales_log.rb

```diff
@@ -348,12 +364,48 @@ def beds_for_la_sale_range
 
   def ownership_scheme
     case ownershipsch
-    when 1
-      "shared ownership"
-    when 2
-      "discounted ownership"
-    when 3
-      "outright sale"
+    when 1 then "shared ownership"
+    when 2 then "discounted ownership"
+    when 3 then "outright sale"
+    end
+  end
+
+private
+
+  def reset_invalidated_dependent_fields!
+    super
+
+    reset_derived_questions!
+  end
+
+  def reset_derived_questions!
+    dependencies = [
+      {
+        parent_conditions: [
+          { attribute: :buylivein, value: 2 },
+        ],
+        derived_attributes: %i[buy1livein buy2livein],
+      },
+      {
+        parent_conditions: [
+          { attribute: :buylivein, value: 1 },
+          { attribute: :jointpur, value: 1 },
+        ],
+        derived_attributes: [:buy1livein],
+      },
+    ]
+
+    dependencies.each do |dependency|
+      any_parent_attributes_changed = dependency[:parent_conditions].any? { |parent_condition| send("#{parent_condition[:attribute]}_changed?") }
```

Comment:
```suggestion
      any_parent_attributes_changed = dependency[:parent_conditions].any? { |attribute, _value| send("#{attribute}_changed?") }
```
Would that work?

Reply:
Also, maybe just conditions instead of parent_conditions?

Reply:
changes made. great shout